Sources/MapboxDirectionsCLI/CodingOperation.swift

```diff
@@ -18,7 +18,9 @@ class CodingOperation<ResponceType : Codable, OptionsType : DirectionsOptions >
         return try encoder.encode(result)
     }
     
-    private func processOutput(_ data: Data) throws {
+    private func processOutput(_ data: Data,
+                               routeResponse: RouteResponse? = nil,
+                               matchResponse: MapMatchingResponse? = nil) throws {
```

Comment:
It's possible to reduce code duplication with this trick:
```swift
protocol DirectionsResultsProvider {
    var directionsResults: [DirectionsResult]? { get }
}

extension RouteResponse: DirectionsResultsProvider {
    var directionsResults: [DirectionsResult]? { routes }
}

extension MapMatchingResponse: DirectionsResultsProvider {
    var directionsResults: [DirectionsResult]? { matches }
}
```
And then `processOutput` can work with `DirectionsResultsProvider` as `shape`, `expectedTravelTime` and `distance` are parts of `DirectionsResult` that is common for both `Route` and `Match`.

And even more, we have `ResponceType` generic type in `CodingOperation` that is either `RouteResponse` or `MapMatchingResponse` and we can mark `ResponceType: DirectionsResultsProvider` and use it directly. `processResponse` already decodes `ResponceType` so we can reuse it.